src/WP_CLI/NonExistentKeyException.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace WP_CLI;
+
+class NonExistentKeyException extends \Exception {
```

Comment:
This should extend one of the [exceptions provided by the SPL](http://php.net/manual/de/spl.exceptions.php).

I think the [`OutOfBoundsException`](http://php.net/manual/de/class.outofboundsexception.php) should be a good fit.